test/components/form/new.spec.js

```diff
@@ -96,15 +96,28 @@ describe('FormNew', () => {
 
   for (const [selectFile, title] of FILE_SELECTION_METHODS) {
     describe(title, () => {
-      it('clicking create button during file read shows info message', () =>
-        Promise.resolve(mountAndMark(FormNew))
+      it('clicking create button during file read shows info message', () => {
+        let originalFunction = null;
+        return Promise.resolve(mountAndMark(FormNew))
           .then(createForm)
+          .then(modal => {
+            originalFunction = FileReader.prototype.readAsText;
+            FileReader.prototype.readAsText = function readAsText() {
+              this.onloadstart();
+            };
+            return modal;
```

Comment:
I can definitely try to run with `const`-ing `originalFunction`.

I can see how modifying the prototype feels a little dangerous... At this point, I haven't come up with any alternatives. I read about modifying `FileReader.prototype` for testing at http://nerds.intuo.io/2016/05/12/mocking-file-uploads-in-javascript.html, though I modified the prototype directly rather than using sinon to do so. Do you think modifying `FileReader.prototype` is reasonably safe in this case? Alternatively, this test is the only time we have a pattern like this, and it’s a test of a bit of an edge case anyway, so part of me feels like another option that wouldn’t be unreasonable would be just to remove the test.

Reply:
I discussed this test with @lognaturel earlier today. I'm going to try to implement slightly different functionality here. Right now, as the file is being read, if you click the Create button, you see an info alert. Instead, I'm going to change it so that as the file is being read, the Create button is disabled. Since the test won't require a click of the Create button, hopefully the file read will still be in progress when the assertion is made, and it won't be necessary to modify the `FileReader` prototype. (There's a chance though that the file read will still be too quick, and we'll have to return to some of these questions about modifying the prototype.) I'll let you know how it goes!

Reply:
FYI, since they both touch alerts, I'm thinking of waiting until we merge #104 before updating this PR.